### DISCUSSION TITLE
Problem creating a composite key

### DISCUSSION THREAD

**Discussion by wcrisafu** [oqtane/oqtane.framework#2009] [Q&A] [answered]
hi,

I am trying to create a few tables linked by a table with a composite key and I keep getting an error. 
I have a table with schools, a table with different types of martial arts and a table to link the many to many relationship. The primary key I am trying to create is:
        private readonly PrimaryKey<TMSSchoolMartialArtEntityBuilder> _primaryKey = new("PK_TMSSchoolMartialArt", x => new { x.TMSSchoolId, x.TMSMartialArtId });

I get an error when I try to run the migration and so far I have been unable to troubleshoot. Any ideas?

**Comment by sbwalker** [comment, marked as answer]
I would suggest creating a standard id identity column for the join table and make it the PK. Your composite fields would be declared as a unique IX but not the PK. And you would create FKs to each related table. 

As a side note you can look at the log file in the /Content/Log folder to see detailed error messages related to migrations.

**Reply by wcrisafu** [reply]
Thank you Shaun. I will try your suggestion.
